keen/api.py

```diff
@@ -179,13 +167,14 @@ def delete_events(self, event_collection, params):
         self._error_handling(response)
         return True
 
+    @requires_key(KeenKeys.READ)
     def get_collection(self, event_collection):
         """
         Extracts info about a collection using the Keen IO API. A master key must be set first.
 
         :param event_collection: the name of the collection to retrieve info for
         """
-        self._check_for_master_key()
+
         url = "{0}/{1}/projects/{2}/events/{3}".format(self.base_url, self.api_version,
                                                        self.project_id, event_collection)
         headers = utilities.headers(self.master_key)
```

Comment:
Oops change to `self.read_key` here and in get_collections() to match the actual key required (now `KeenKeys.READ`).